crates/ironrdp-client/src/network_client.rs

```diff
@@ -0,0 +1,116 @@
+use sspi::{Error, ErrorKind};
+use std::net::{IpAddr, Ipv4Addr};
+use std::{future::Future, pin::Pin};
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::{TcpStream, UdpSocket};
+
+use ironrdp::connector::{custom_err, general_err, ConnectorResult};
+use reqwest::Client;
+use url::Url;
+
+use ironrdp_tokio::AsyncNetworkClient;
+pub(crate) struct AsyncTokioNetworkClient {
```

Comment:
This is probably a more accurate name:
```suggestion
pub(crate) struct ReqwestNetworkClient {
```

Reply:
updated